tests/test_paragraphs.py

```diff
@@ -0,0 +1,84 @@
+
+                # Remove annoying string sequences.
+                # - Multiline code sections.
+                content = re.sub(r'```.+?(?=```)```', '\n', content, flags=re.S)
+                content = re.sub(r'\n`.+?(?=`\n)`\n', '\n', content, flags=re.S)
+                # - Notes.
+                content = re.sub(r'\n\s*>.+?(?=\n\n)', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*>.+?(?=\n$)', '\n', content, flags=re.S)
+                # - Showdown extensions.
+                content = re.sub(r'%figure.+?(?=%end)%end', '\n', content, flags=re.S)
+                content = re.sub(r'%api.+?(?=%end)%end', '\n', content, flags=re.S)
+                # - Headers.
+                content = re.sub(r'^#.*', '\n', content)
+                content = re.sub(r'\n#.*', '\n', content)
+                # - Items.
+                content = re.sub(r'\n\s*-.+?(?=\n\n)', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*-.+?(?=\n$)', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*\d+\..+?(?=\n\n)', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*\d+\..+?(?=\n$)', '\n', content, flags=re.S)
+                content = re.sub(r'\n    .+?(?=\n)', '\n', content, flags=re.S)
+                content = re.sub(r'\n        .+?(?=\n)', '\n', content, flags=re.S)
+                # - HTML statements
+                content = re.sub(r'\n<.+?>\n', '\n', content, flags=re.S)
+                content = re.sub(r'\n---\n', '\n', content, flags=re.S)
+                # - Single hyperlinks
+                content = re.sub(r'\n\!?\[.+\)\n', '\n', content, flags=re.S)
+                # - Special statements
+                content = re.sub(r'\nRelease {{.+?}}\n', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*\*\*.+?\n', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*\*.+?\*\n', '\n', content, flags=re.S)
+                content = re.sub(r'\n\s*\{.+?\n', '\n', content, flags=re.S)
+
+                # Extract paragraphs.
+                for match in re.finditer(r'(?s)((?:[^\n][\n]?)+)', content):
+                    paragraph = content[match.start():match.end() - 1]
+                    # - Arrays.
+                    if paragraph.startswith('| ') or paragraph.startswith('> '):
+                        continue
+                    self.paragraphs.append({'paragraph': paragraph, 'md': md_path})
+        # Debug: Uncomment to display all the acquired paragraphs.
+        # for p in self.paragraphs:
+        #     print ('@@@')
+        #     print (p)
+
+    def test_one_sentence_per_line(self):
+        """Test that each sentence is written on one line."""
+        for p in self.paragraphs:
+            lines = p['paragraph'].split('\n')
+            for line in lines:
+                line = line.strip()
+                if len(line) == 0:
+                    continue
+                if '**Keywords**' in line:
```

Comment:
Would be nice to write a small comment explaining why we skip this line.

Reply:
Even simpler, this is no more required. Indeed, some special lines are acting like titles and so it's not required to test that these titles are ending with points, etc. But this is now filtered out correctly in the test setup.